scripts/tests/jobs/smn/perpetuation.lua

```diff
@@ -0,0 +1,229 @@
+---@param player CTestPuppet
+---@param cost integer
+local function verifyPerpetuationCost(player, cost)
+    player:setMP(player:getMaxMP())
+    local petName = player:getPetName() or 'avatar'
+    for i = 1, 5 do
+        local startMP = player:getMP()
+        xi.test.world:tick(xi.tick.EFFECT)
+        local lostMP = startMP - player:getMP()
+        assert(
+            lostMP == cost,
+            string.format('Expected to lose %u with %s summoned but lost %u.',
+                cost, petName, lostMP)
+        )
+    end
+end
+
+local specialBaseCost =
+{
+    [xi.petId.CARBUNCLE] = 9,
+    [xi.petId.FENRIR] = 11,
+}
+
+describe('Avatar perpetuation', function()
+    ---@type CTestPuppet
+    local player
+
+    before_each(function()
+        player = xi.test.world:spawnPlayer({ zone = xi.zone.WEST_RONFAURE, job = xi.job.SMN, level = 75 })
+        player:setMod(xi.mod.REFRESH, 0)
+        xi.test.world:tick()
+    end)
+
+    it('has normal cost without matching day or weather', function()
+        -- Verify no gain or loss of MP without avatar
+        verifyPerpetuationCost(player, 0)
+        -- Verify losing MP according to avatar perpetuation
+        xi.test.world:setVanaDay(xi.day.DARKSDAY)
+        player:spawnPet(xi.petId.GARUDA)
+        verifyPerpetuationCost(player, 13)
+        player:despawnPet()
+        player:spawnPet(xi.petId.CARBUNCLE)
+        verifyPerpetuationCost(player, 9)
+    end)
```

Comment:
and then presumably the whole world comes down and all the entities and clients are cleaned up?